### ISSUE TITLE
Audio sent ignores channels

### ISSUE THREAD

**Issue by mlugg** [vktec/vmv#1]
Audio sent by sound\_io.c ignores the amount of channels in the input audio stream, instead simply sending all channels consecutively. A better solution would be to average the audio across all channels or to deal with this in audio.c.

**Comment by mlugg** [comment]
I was unaware of this. That works, however, if you have non-stereo audio (I can't set the amount of channels in libsoundio) it'll still look fine, (except sub might look odd), but I'd rather it was dealt with properly :]

**Comment by vktec** [comment]
Yeah, probably a good plan. Is the channel info exposed somehow?

**Comment by mlugg** [comment]
Yeah, you can get it from the soundinfo struct passed to audio.c - I'll find the appropriate members in a moment. Shall I leave the soundinfo code as is then, which just sends all channels consecutively?

**Comment by mlugg** [comment]
http://libsound.io/doc-1.0.0/structSoundIoChannelLayout.html - that can be found in the `sinfo` parameter passed to audio.c, specifically `sinfo->instream->layout`. You probably only need `sinfo->instream->layout.channel_count` but I guess you could do something more sophisticated if you wanted :P

**Comment by vktec** [comment]
Yes, leave the soundio stuff alone. I'll handle averaging the channels in `audio.c`

**Comment by vktec** [comment]
I think I might've misremembered...

Seems like I had meant to do the channel averaging, forgotten, and then remembered my meaning to do it and assumed I'd done it.

TL;DR it never happened in the first place.
